npm/npm.go

```diff
@@ -77,6 +80,58 @@ func (npMgr *NetworkPolicyManager) GetClusterState() telemetry.ClusterState {
 	return npMgr.clusterState
 }
 
+// SendAiMetrics :- send NPM metrics using AppInsights
+func (npMgr *NetworkPolicyManager) SendAiMetrics() {
+	var (
+		aiConfig = aitelemetry.AIConfig{
+			AppName:                   util.AzureNpmFlag,
+			AppVersion:                npMgr.version,
+			BatchSize:                 32768,
+			BatchInterval:             30,
+			RefreshTimeout:            15,
+			DebugMode:                 true,
+			GetEnvRetryCount:          5,
+			GetEnvRetryWaitTimeInSecs: 3,
+		}
+		th, err = aitelemetry.NewAITelemetry("", aiMetadata, aiConfig)
+		heartbeat = time.NewTicker(time.Minute * heartbeatIntervalInMinutes).C
+		customDimensions = map[string]string{"ClusterID": util.GetClusterID(npMgr.nodeName),
+												"APIServer": npMgr.serverVersion.String()}
+		podCount = aitelemetry.Metric{
+			Name: "PodCount",
+			CustomDimensions: customDimensions,
+		}
+		nsCount = aitelemetry.Metric{
+			Name: "NsCount",
+			CustomDimensions: customDimensions,
+		}
+		nwPolicyCount = aitelemetry.Metric{
+			Name: "NwPolicyCount",
+			CustomDimensions: customDimensions,
+		}
+	)
+
+	for ; err != nil; {
```

Comment:
why are we not checing this error(at 97)  before defining all those metrics?

Reply:
because that's in a var block